File: binds/go.v1/src/webbrowser.go
package main

import (
	"fmt"
	"time"

	"gopkg.in/inconshreveable/log15.v2"
	"gopkg.in/project-iris/iris-go.v1"
)

func init() {
	iris.Log.SetHandler(log15.DiscardHandler())
}

func main() {
	// START OMIT

// Connect to the network as a simple client
connection, err := iris.Connect(55555) // HLreq
if err != nil {
	fmt.Println("Failed to connect:", err); return
}
defer connection.Close() // HLreq

// Issue a dummy request every second
for i := 1; i <= 60; i++ {
	request := []byte(fmt.Sprint("Request #", i))

	reply, err := connection.Request("webserver", request, time.Second) // HLreq
	if err != nil {
		fmt.Println("Request failed:", err)
	} else {
		fmt.Println("Web reply:", string(reply))
	}
	time.Sleep(time.Second)
}
// END OMIT
}

File: binds/erlang.v1/src/webbrowser.erl
-module(webbrowser).
-export([main/0]).

main() ->
%% START OMIT
% Connect to the Iris network as a simple client
{ok, Client} = iris_client:start(55555), // HLreq

% Issue a dummy request every second
lists:foreach(fun(Index) ->
    Request = iolist_to_binary(io_lib:format("Request #~p", [Index])),
    case iris_client:request(Client, "webserver", Request, 1000) of // HLreq
        {ok, Reply}     -> io:format("Web reply: ~s~n", [Reply]);
        {error, Reason} -> io:format("Request failed: ~s~n", [Reason])
    end,
    timer:sleep(1000)
end, lists:seq(1, 100)),

% Disconnect from the Iris network
ok = iris_client:stop(Client). // HLreq
%% END OMIT
